tests/test_browser.py

```diff
@@ -81,20 +80,25 @@ def test__request(httpbin):
         "Content-Type"]
 
 
-def test__request_file(httpbin):
+@pytest.mark.parametrize('expected_content, set_value', [
+    (":-)", True),
+    ("", False)
+])
+def test__request_file(httpbin, expected_content, set_value):
     form_html = """
     <form method="post" action="{}/post">
       <input name="pic" type="file" />
     </form>
     """.format(httpbin.url)
     form = BeautifulSoup(form_html, "lxml").form
 
-    # create a temporary file for testing file upload
-    pic_path = tempfile.mkstemp()[1]
-    with open(pic_path, "w") as f:
-        f.write(":-)")
+    if set_value:
+        # create a temporary file for testing file upload
+        pic_path = tempfile.mkstemp()[1]
```

Comment:
This will leave a temporary file around. Perhaps we should delete it at the end of this test (or use [NamedTemporaryFile](https://docs.python.org/3.7/library/tempfile.html#tempfile.NamedTemporaryFile) instead if it fits into the logic in a nice way, which it might not due to the `if set_value` statement).

Reply:
Pytest also has some temporary directory fanciness, as another option for you! https://docs.pytest.org/en/latest/tmpdir.html

Reply:
Yes ! I completely missed it !
I'll try to add a simple file deletion before ending the test